datalake-spark3/src/main/scala/bio/ferlab/datalake/spark3/loader/Loader.scala

```diff
@@ -46,6 +46,27 @@ trait Loader {
                          format: String,
                          options: Map[String, String] = Map("dataChange" -> "true"))(implicit spark: SparkSession): DataFrame
 
+  /**
+   * Dynamically keeps old partition and overwrites new partitions.
+   *
+   * @param location     where to write the data
+   * @param databaseName database name
+   * @param tableName    table name
+   * @param df           new data to write into the table
+   * @param partitioning partition columns
+   * @param format       format
+   * @param options      write options
+   * @param spark        a spark session
+   * @return updated data
+   */
+  def overwritePartitionDynamic(location: String,
```

Comment:
Do you really need to define this method in Loadder interface? Could be directly a member of DeltaLoader